cli/provider/core_service_linux.go

```diff
@@ -0,0 +1,133 @@
+//go:build linux
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"go.keploy.io/server/v2/config"
+	"go.keploy.io/server/v2/pkg/core"
+	"go.keploy.io/server/v2/pkg/core/hooks"
+	"go.keploy.io/server/v2/pkg/core/proxy"
+	"go.keploy.io/server/v2/pkg/core/tester"
+	"go.keploy.io/server/v2/pkg/models"
+	"go.keploy.io/server/v2/pkg/platform/docker"
+	"go.keploy.io/server/v2/pkg/platform/telemetry"
+	"go.keploy.io/server/v2/pkg/platform/yaml/configdb/testset"
+	mockdb "go.keploy.io/server/v2/pkg/platform/yaml/mockdb"
+	reportdb "go.keploy.io/server/v2/pkg/platform/yaml/reportdb"
+	testdb "go.keploy.io/server/v2/pkg/platform/yaml/testdb"
+	"go.keploy.io/server/v2/pkg/service/record"
+	"go.keploy.io/server/v2/pkg/service/replay"
+	"go.keploy.io/server/v2/utils"
+	"go.uber.org/zap"
+)
+
+type CommonInternalService struct {
```

Comment:
what could be a good 1 word name for this?

Reply:
may be MicroServices